main-01.py

```diff
@@ -136,8 +135,7 @@ def process_songs():
 
  # beat_key=name+".txt"
     #filt and get all mp3 files in local directory
-  all_song_keys=[]
-  get_all_song_keys(all_song_keys)
+  all_song_keys = get_all_song_keys()
```

Comment:
You do not want to pass a variable into a function and then have the function change it (mutate it).  What I did is change `get_all_song_keys` slightly so it just returns all the song keys. 

What you were doing before is starting with a `all_song_keys` list and then giving that list to the function to fill with stuff.  Giving the list to a function is not as clean, because as the function you pass the list to becomes more complex, it will become hard to predict and understand what the function does to that list and what kind of elements it's putting into that list.  

I can tell you more about this sometime.  You can also look up "pass argument by reference" to read about passing arguments by reference.  Also read about the benefits of "functional programming".